README.md

```diff
@@ -13,10 +14,11 @@ sorted, i.e. do not copy the list and sort the copy.
 
 ## Runtime Analysis
 
-What is the runtime complexity of the algorithm that you implemented? What does
-a best case input for your implementation look like, what does a worst case
-input look like? How would this complexity change if you generated permutations
-randomly without memory instead of systematically trying them?
+Best Case:Best Case: If the input array is already sorted, the algorithm will terminate immediately after the first call to isSorted(). Hence, the best-case time complexity is O(n), where n is the number of elements in the array.
 
-Describe your reasoning and the conclusion you've come to. Your reasoning is the
-most important part. Add your answer to this markdown file.
+The worst-case scenario occurs when the input array is in reverse sorted order. In this case, the algorithm will generate all permutations of the array, and it needs to check each permutation for sorting. Generating all permutations and checking each one for sorting would lead to a time complexity of O(n! (n))
+
+Impact of Random Generation without Memory:
+If permutations were generated randomly without memory, the time complexity would still be O(n! (n)) in the worst case because the algorithm would still need to consider all possible permutations.
```

Comment:
Is there any guarantee that this algorithm would ever terminate?

Reply:
Since the algorithm generates permutations recursively and checks each permutation for sorting, it will eventually terminate even for large input arrays. However, the time it takes to terminate can grow rapidly, especially as the size of the input array increases.

Reply:
What if permutations are generated randomly?

Reply:
Kind of like the monkey in front of typewriter will eventually write Shakespeare

Reply:
But is there a guarantee?

Reply:
I believe so, but the sun could explode before it happens

Reply:
Ok, put another way -- if you generate random numbers, is there a guarantee that it will ever be exactly 3.1415?

Reply:
Im still confused, given infinite time yes, but I suppose there's not a 100% guarantee

Reply:
Ok, so the algorithm might never terminate?

Reply:
I suppose the algorithm might never terminate

Reply:
So the complexity would not be what you say above.

Reply:
Theres a chance it terminates very quickly, but an equally probable cause where it never terminates. Im not really sure how to compute a worst case as the worst case would just be infinite time

Reply:
You can just say that.